### PR TITLE
This adds forms for create, update and delete for author and book

### PR DESCRIPTION
**Pull request by hamadhassan3**
# Overview of the work
This project is part of the 3rd week in Django training program. It implements a website based on the Django web framework.

# Major Changes

1. A form for renewing book
2. Forms for create, update and delete for author
3. Forms for create, update and delete for book

# Output

## Librarian can renew the books now
output1

## Renew form
output2

## Form for creating book
output3

## Form for creating author
output4

# Feedback Required
I need feedback on the quality and style of code. Please also mention if any part of the code can be improved.

### CODE REVIEW

File: catalog/forms.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+from django.core.exceptions import ValidationError
+from django.utils.translation import ugettext_lazy as _
+
+
+
+
+class RenewBookForm(forms.Form):
+    renewal_date = forms.DateField(help_text="Enter a date between now and 4 weeks (default 3).")
+
+    def clean_renewal_date(self):
```
Comment (shahabejaz) [inline catalog/forms.py]: Doc string and type hint would be useful here.

File: catalog/views.py [hunk 2]
```diff
@@ -1,8 +1,19 @@
...
+from django.shortcuts import get_object_or_404
+from django.http import HttpResponseRedirect
+from django.urls import reverse
+from catalog.forms import RenewBookForm
+from django.views.generic.edit import CreateView, UpdateView, DeleteView
+from django.urls import reverse_lazy
+from catalog.models import Author
+
 
 def index(request):
```
Comment (shahabejaz) [inline catalog/views.py]: I would add type hint here.

File: catalog/views.py [hunk 3]
```diff
@@ -120,4 +131,70 @@ class AllLoanedBooks(PermissionRequiredMixin, generic.ListView):
     def get_queryset(self):
```
Comment (shahabejaz) [inline catalog/views.py]: Add type hint here.

File: catalog/views.py [hunk 4]
```diff
@@ -120,4 +131,70 @@ class AllLoanedBooks(PermissionRequiredMixin, generic.ListView):
...
+        return BookInstance.objects.filter(status__exact='o').order_by('due_back')
+
+
+
+
+
+
+@login_required
+@permission_required('catalog.can_mark_returned', raise_exception=True)
+def renew_book_librarian(request, pk):
```
Comment (shahabejaz) [inline catalog/views.py]: I would add type hint here.

File: catalog/forms.py (hunk 1 above)
Comment (hamadhassan3) [reply catalog/forms.py]: The docstring and type hint are added now

File: catalog/views.py (hunk 2 above)
Comment (hamadhassan3) [reply catalog/views.py]: Added now

File: catalog/views.py (hunk 3 above)
Comment (hamadhassan3) [reply catalog/views.py]: Added now

File: catalog/views.py (hunk 4 above)
Comment (hamadhassan3) [reply catalog/views.py]: Added now

**Review by shahabejaz — APPROVED** [review]

**Pull request merged**